components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/AccountConfirmationValidationHandler.java

```diff
@@ -115,17 +121,18 @@ public void handleEvent(Event event) throws IdentityEventException {
                         IdentityCoreConstants.USER_ACCOUNT_NOT_CONFIRMED_ERROR_CODE);
                 IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
                 throw new IdentityEventException(IdentityCoreConstants.USER_ACCOUNT_NOT_CONFIRMED_ERROR_CODE,
-                        "User : " + userName + " not confirmed yet.");
+                        "User : " + Utils.maskIfRequired(userName) + " not confirmed yet.");
             } else if (isInvalidCredentialsScenario(operationStatus, user)) {
                 if (log.isDebugEnabled()) {
                     log.debug(String.format("Account unconfirmed user: %s in userstore: %s in tenant: %s is trying " +
-                            "to log in with an invalid password", userName, domainName, tenantDomain));
+                                    "to log in with an invalid password", Utils.maskIfRequired(userName), domainName,
```

Comment:
Let's remove the masking as this is a debug log.

Reply:
Resolved with https://github.com/wso2-extensions/identity-governance/pull/821/commits/f6105f68b792402a2ae230387d8ed7c44dcaa089